src/docfx/lib/log/Telemetry.cs

```diff
@@ -73,6 +74,28 @@ public static void SetRepository(string? repo, string? branch)
             s_telemetryClient.Context.GlobalProperties["Branch"] = s_branch;
         }
 
+        public static void TrackDocfxConfig(string fileName, string docsetName, string configContent)
+        {
+            var docfxConfigTelemetryValue = configContent;
+            try
+            {
+                var source = new FilePath(fileName);
+                var (_, config) = fileName.EndsWith(".yml", PathUtility.PathComparison)
+                    ? YamlUtility.Parse(configContent, source)
+                    : JsonUtility.Parse(configContent, source);
+                docfxConfigTelemetryValue = JsonUtility.Serialize(JsonUtility.ToObject<DocfxConfigTelemetryModel>(config));
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error happens when deserializing the config file {fileName}: {ex.Message}");
+            }
+
+            var dimensions = new Dictionary<string, string>();
+            dimensions["DocsetName"] = docsetName;
+            dimensions["Config"] = docfxConfigTelemetryValue;
+            TrackEvent(fileName, dimensions);
```

Comment:
nit: why not directly call `s_telemetryClient.TrackEvent`? this helper method does not seem to add much value.

Reply:
this method is used to construct the eventTelemetry?

---
In reply to: [452600406](https://github.com/dotnet/docfx/pull/6216#discussion_r452600406) [](ancestors = 452600406)